Classes/Emogrifier.php

```diff
@@ -1529,18 +1536,32 @@ private function getNodesToExclude(\DOMXPath $xPath)
     public function handleXpathError($type, $message, $file, $line, array $context)
     {
         if ($type === E_WARNING && isset($context['cssRule']['selector'])) {
-            throw new \InvalidArgumentException(
-                sprintf(
-                    '%s in selector >> %s << in %s on line %s',
-                    $message,
-                    $context['cssRule']['selector'],
-                    $file,
-                    $line
-                )
-            );
+            if ($this->debug) {
+                throw new \InvalidArgumentException(
+                    sprintf(
+                        '%s in selector >> %s << in %s on line %s',
+                        $message,
+                        $context['cssRule']['selector'],
+                        $file,
+                        $line
+                    )
+                );
+            } else {
+                return false;
+            }
         }
 
         // the normal error handling continues when handler return false
         return false;
     }
+
+    /**
+     * Set debug mode.
```

Comment:
Please always use the third person for the first sentence of a method description: [This Method] "Sets the debug mode."